sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -305,7 +305,9 @@ def receive(self, max_batch_size=None, max_wait_time=None):
                 :caption: Receive messages from ServiceBus.
 
         """
-        self._can_run()
+        if max_batch_size and self._config.prefetch < max_batch_size:
+            raise ValueError("max_batch_size should be <= prefetch of ServiceBusClient.")
```

Comment:
I would add more detailed instruction here:

 `ValueError("max_batch_size should be less equal than prefetch of ServiceBusReceiver or you could set the prefetch parameter when you're constructing the ServiceBusReceiver.")`

(BTW, it's ServiceBusReceiver not ServiceBusClient)